src/main/java/com/exactpro/th2/common/schema/factory/AbstractCommonFactory.java

```diff
@@ -350,15 +350,19 @@ public GrpcRouter getGrpcRouter() {
             Class<T> messageClass,
             MessageConverter<T> messageConverter,
             Set<String> defaultSendAttributes,
-            Set<String> defaultSubscribeAttributes
+            Set<String> defaultSubscribeAttributes,
+            String... labels
     ) {
         customMessageRouters.compute(
                 messageClass,
                 (msgClass, curValue) -> {
                     if (curValue != null) {
                         throw new IllegalStateException("Message router for type " + msgClass.getCanonicalName() + " is already registered");
                     }
-                    var router = new RabbitCustomRouter<>(msgClass.getSimpleName(), messageConverter, defaultSendAttributes,
+                    if (labels == null) {
+                        throw new IllegalArgumentException("No labels provided for message router with type" + msgClass.getCanonicalName());
+                    }
```

Comment:
Objects.requireNonNull(lebels, () -> "Lables cannot be null for custom message router")